app/models/notifications/private_message.rb

```diff
@@ -1,15 +1,29 @@
-class Notifications::PrivateMessage < Notification
-  def mail_job
-    Workers::Mail::PrivateMessage
-  end
-  def popup_translation_key
-    'notifications.private_message'
-  end
-  def self.make_notification(recipient, target, actor, notification_type)
-    n = notification_type.new(:target => target,
-                               :recipient_id => recipient.id)
-    target.increase_unread(recipient)
-    n.actors << actor
-    n
+module Notifications
+  class PrivateMessage < Notification
+    def mail_job
+      Workers::Mail::PrivateMessage
+    end
+
+    def popup_translation_key
+      "notifications.private_message"
+    end
+
+    def self.notify(object, _recipient_user_ids)
+      case object
+      when Conversation
+        object.messages.each {|message| notify_message(message) }
+      when Message
+        notify_message(object)
+      end
+    end
+
+    def self.notify_message(message)
+      recipients = message.conversation.participants.select(&:local?) - [message.author]
+      recipients.map(&:owner).each do |recipient|
+        message.increase_unread(recipient)
+        new(recipient: recipient).email_the_user(message, message.author)
+      end
```

Comment:
Same story about loading the entire collections, I'll skip any further ones for now.